cmd/celestia-da/main.go

```diff
@@ -31,9 +40,57 @@ func main() {
 	rootCmd := &cobra.Command{
 		Use:   "celestia-da",
 		Short: "Celesia DA layer gRPC server for rollkit",
-		Run: func(cmd *cobra.Command, args []string) {
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
At this point, we should consider importing the command directly from the node, as its looks like 1:1 copy

Reply:
We need to inject this call which spins up the grpc service into the command `RunE` function: https://github.com/rollkit/celestia-da/blob/main/cmd/celestia-da/main.go#L61-L90

Reply:
If you need it in the end, you could add `PostRunE` field to it which runs serve

Reply:
Definitely worth trying out `PostRunE` :+1:

Reply:
`PostRunE` seems to be blocked by `Run`. Also needed to query flags, so went with overriding `start.RunE`